src/state/nftMarket/helpers.ts

```diff
@@ -0,0 +1,29 @@
+import { request, gql } from 'graphql-request'
+import { GRAPH_API_NFTMARKET } from 'config/constants/endpoints'
+
+export const getCollections = async (): Promise<any[]> => {
+  try {
+    const res = await request(
+      GRAPH_API_NFTMARKET,
+      gql`
+        {
+          collections {
+            id
+            name
+            symbol
+            active
+            totalVolumeBNB
+            numberTokensListed
+            tradingFee
+            creatorFee
+          }
+        }
+      `,
+    )
+
+    return res.collections
+  } catch (error) {
+    console.error('Failed to fetch NFT collections', error)
+    return null
```

Comment:
As the collections state is typed as an array we may end up calling array funcs on a null val and cause a crash - is it worth returning this as an empty array?